packages/odyssey-react-mui/src/Tabs.tsx

```diff
@@ -100,16 +111,51 @@ const Tabs = ({
   return (
     <MuiTabContext value={tabState}>
       <MuiTabList onChange={onChange} aria-label={ariaLabel}>
-        {tabs.map((tab, index) => (
-          <MuiTab
-            data-se={tab.testId}
-            disabled={tab.isDisabled}
-            icon={tab.startIcon}
-            label={tab.label}
-            value={tab.value ? tab.value : index.toString()}
-            key={tab.value ? tab.value : index.toString()}
-          />
-        ))}
+        {tabs.map((tab, index) => {
+          const {
+            testId,
+            isDisabled,
+            label,
+            startIcon,
+            value,
+            badgeContent,
+            badgeContentMax,
+          } = tab;
+
+          const BadgeLabel = () => {
+            return (
+              <>
+                {label}
+                {!!badgeContent && badgeContent > 0 && (
+                  <Box
+                    sx={{
+                      marginInlineStart: badgeContent
+                        ? odysseyDesignTokens.Spacing2
+                        : 0,
+                    }}
+                  >
+                    <Badge
+                      badgeContent={badgeContent}
+                      badgeContentMax={badgeContentMax}
+                      type={value === tabState ? "attention" : "default"}
+                    />
+                  </Box>
+                )}
+              </>
+            );
+          };
```

Comment:
My preference is to see some extraction of how 2 components are generated to outside the body of the render function:
```

function TabWithBadge(index, props) {
   const { /* extract props */ } = props;
   return (
    <MuiTab
              data-se={testId}
              disabled={isDisabled}
              icon={startIcon}
              label={<BadgeLabel { ...props } />}
              value={value ? value : index.toString()}
              key={value ? value : index.toString()}
            />
  );
}


function BadgeLabel(props) {
  return (
    <>
                {label}
                {!!badgeContent && badgeContent > 0 && (
                  <Box
                    sx={{
                      marginInlineStart: badgeContent
                        ? odysseyDesignTokens.Spacing2
                        : 0,
                    }}
                  >
                    <Badge
                      badgeContent={badgeContent}
                      badgeContentMax={badgeContentMax}
                      type={value === tabState ? "attention" : "default"}
                    />
                  </Box>
                )}
              </>
  );
}
```

Then your actual code in the render itself would be something like:

```
const tabsWithBages = tabs.map((tab, index) => <TabWithBadge {...tab} />)

return (
  <MuiTabList onChange={onChange} aria-label={ariaLabel}>
    {tabsWithBadges}
  </MuiTabList>
)
```

My motivation is to make the render function leaner, more performant and easier to read.
You may be able to memoize the components as well to save additional renders on prop updates.